File: app/src/main/java/wangdaye/com/geometricweather/settings/dialogs/MinimalIconDialog.java
package wangdaye.com.geometricweather.settings.dialogs;

import android.annotation.SuppressLint;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.appcompat.widget.AppCompatImageView;

import wangdaye.com.geometricweather.R;
import wangdaye.com.geometricweather.common.basic.GeoDialog;
import wangdaye.com.geometricweather.common.basic.models.weather.WeatherCode;
import wangdaye.com.geometricweather.theme.resource.ResourceHelper;
import wangdaye.com.geometricweather.theme.resource.providers.DefaultResourceProvider;
import wangdaye.com.geometricweather.theme.resource.providers.ResourceProvider;
import wangdaye.com.geometricweather.theme.resource.ResourcesProviderFactory;

/**
 * Adaptive icon dialog.
 * */
public class MinimalIconDialog extends GeoDialog {

    private static final String KEY_WEATHER_CODE = "weather_code";
    private static final String KEY_DAYTIME = "daytime";
    private static final String KEY_RESOURCE_PROVIDER = "resource_provider";

    public static MinimalIconDialog getInstance(WeatherCode code,
                                                boolean daytime,
                                                ResourceProvider provider) {
        Bundle bundle = new Bundle();
        bundle.putString(KEY_WEATHER_CODE, code.name());
        bundle.putBoolean(KEY_DAYTIME, daytime);
        bundle.putString(KEY_RESOURCE_PROVIDER, provider.getPackageName());

        MinimalIconDialog dialog = new MinimalIconDialog();
        dialog.setArguments(bundle);
        return dialog;
    }

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater,
                             @Nullable ViewGroup container,
                             @Nullable Bundle savedInstanceState) {
        super.onCreateView(inflater, container, savedInstanceState);

        View view = LayoutInflater.from(getActivity()).inflate(
                R.layout.dialog_minimal_icon, container, false);
        initWidget(view);
        return view;
    }

    @SuppressLint("SetTextI18n")
    private void initWidget(View view) {
        Bundle bundle = getArguments();
        if (bundle == null) {
            return;
        }

        String codeName = bundle.getString(KEY_WEATHER_CODE);
        WeatherCode code = codeName == null ? WeatherCode.CLEAR : WeatherCode.valueOf(codeName);

        boolean daytime = bundle.getBoolean(KEY_DAYTIME, true);

        String providerPackageName = bundle.getString(KEY_RESOURCE_PROVIDER);
        ResourceProvider provider = providerPackageName == null
                ? new DefaultResourceProvider()
                : ResourcesProviderFactory.getNewInstance(providerPackageName);


        AppCompatImageView xmlIcon = view.findViewById(R.id.dialog_minimal_icon_xmlIcon);
        xmlIcon.setImageDrawable(ResourceHelper.getMinimalXmlIcon(provider, code, daytime));

        TextView titleView = view.findViewById(R.id.dialog_minimal_icon_title);
        titleView.setText(code.name() + (daytime ? "_DAY" : "_NIGHT"));

        AppCompatImageView lightIconView = view.findViewById(R.id.dialog_minimal_icon_lightIcon);
        lightIconView.setImageDrawable(ResourceHelper.getWidgetNotificationIcon(
                provider, code, daytime, true, "light"));

        AppCompatImageView greyIconView = view.findViewById(R.id.dialog_minimal_icon_greyIcon);
        greyIconView.setImageDrawable(ResourceHelper.getWidgetNotificationIcon(
                provider, code, daytime, true, "grey"));

        AppCompatImageView darkIconView = view.findViewById(R.id.dialog_minimal_icon_darkIcon);
        darkIconView.setImageDrawable(ResourceHelper.getWidgetNotificationIcon(
                provider, code, daytime, true, "dark"));
    }
}

File: app/src/main/java/org/breezyweather/ui/settings/dialogs/MinimalIconDialog.kt
/**
 * This file is part of Breezy Weather.
 *
 * Breezy Weather is free software: you can redistribute it and/or modify it
 * under the terms of the GNU Lesser General Public License as published by the
 * Free Software Foundation, version 3 of the License.
 *
 * Breezy Weather is distributed in the hope that it will be useful, but
 * WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
 * or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public
 * License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with Breezy Weather. If not, see <https://www.gnu.org/licenses/>.
 */

package org.breezyweather.ui.settings.dialogs

import android.annotation.SuppressLint
import android.content.Context
import android.view.LayoutInflater
import android.view.View
import androidx.appcompat.widget.AppCompatImageView
import breezyweather.domain.weather.reference.WeatherCode
import com.google.android.material.dialog.MaterialAlertDialogBuilder
import org.breezyweather.R
import org.breezyweather.ui.theme.resource.ResourceHelper
import org.breezyweather.ui.theme.resource.providers.ResourceProvider

object MinimalIconDialog {
    fun show(
        context: Context,
        code: WeatherCode,
        daytime: Boolean,
        provider: ResourceProvider,
    ) {
        val view = LayoutInflater
            .from(context)
            .inflate(R.layout.dialog_minimal_icon, null, false)
        initWidget(view, code, daytime, provider)
        MaterialAlertDialogBuilder(context)
            .setTitle(code.name + if (daytime) "_DAY" else "_NIGHT")
            .setView(view)
            .show()
    }

    @SuppressLint("SetTextI18n")
    private fun initWidget(
        view: View,
        code: WeatherCode,
        daytime: Boolean,
        provider: ResourceProvider,
    ) {
        val lightIconView = view.findViewById<AppCompatImageView>(R.id.dialog_minimal_icon_lightIcon)
        lightIconView.setImageDrawable(
            ResourceHelper.getWidgetNotificationIcon(provider, code, daytime, true, "light")
        )
        val greyIconView = view.findViewById<AppCompatImageView>(R.id.dialog_minimal_icon_greyIcon)
        greyIconView.setImageDrawable(
            ResourceHelper.getWidgetNotificationIcon(provider, code, daytime, true, "grey")
        )
        val darkIconView = view.findViewById<AppCompatImageView>(R.id.dialog_minimal_icon_darkIcon)
        darkIconView.setImageDrawable(
            ResourceHelper.getWidgetNotificationIcon(provider, code, daytime, true, "dark")
        )
    }
}
